src/schemas/TimesheetSchema.tsx

```diff
@@ -1,12 +1,31 @@
 import { z } from "zod";
+import {RowSchema} from './RowSchema'; 
+
+// The status is either undefined, for not being at that stage yet, or 
+// contains the date and author of approving this submission 
+export const StatusEntryType = z.union(
+  [
+  z.object({
+    Date: z.number(), 
+    Author: z.string()
+  }), 
+  z.undefined()]); 
+
+// Status type contains the four stages of the pipeline we have defined 
+export const StatusType = z.object({
+  Employee: StatusEntryType, 
+  Manager: StatusEntryType,
+  Admin: StatusEntryType, 
+  Accepted: StatusEntryType 
+});
 
 export const TimeSheetSchema = z.object({
   TimesheetID: z.number(), 
   UserID: z.string(), 
   StartDate: z.number(),
-  Status: z.string(),
+  Status: StatusType, 
   Company: z.string(), 
-  TableData: z.array(z.any()), 
-})
+  TableData: z.array(RowSchema), 
```

Comment:
I have this as HoursData and then ScheduleData (for supervisor schedule entries) on backend for database, but can merge them together when they get passed to the frontend if that's easiest for you